### DISCUSSION TITLE
EMS-ESP and DS18B20

### DISCUSSION THREAD

**Discussion by bluhminga** [community/community#69678] [Repositories] [answered]
### Select Topic Area

Question

### Body

Hello,

I use EMS-ESP with the "BBQ interface card V3.1". And an "az delivery esp32 wroom 32" module.
Works great in conjunction with Home Assistant.

Is it possible to connect a DS18B20 to the ESP32 module and display it in EMS-ESP? And then send the “room temperature” to Home Assistant?

Thanks in advance.

**Comment by moayad-star** [comment, marked as answer]
Yes, it is possible to connect a DS18B20 temperature sensor to your ESP32 module and display the temperature in EMS-ESP. Here are the general steps to achieve this:

1. **Connect DS18B20 to ESP32**: Wire the DS18B20 sensor to the ESP32 following the sensor's datasheet. Usually, it requires connecting the sensor's data pin to a GPIO pin on the ESP32.

2. **Install Required Libraries**: You will need to install the necessary libraries for both the DS18B20 sensor and the EMS-ESP software. You can usually find these libraries on GitHub or through the Arduino Library Manager.

3. **Write ESP32 Code**: Write a code for your ESP32 to read the temperature data from the DS18B20 sensor. You will need to use the OneWire and DallasTemperature libraries for this purpose.

4. **Display Temperature in EMS-ESP**: Modify the EMS-ESP code to display the temperature data from the DS18B20 sensor. You may need to create a custom sensor entity and update the display in EMS-ESP accordingly.

5. **Send Data to Home Assistant**: To send the room temperature to Home Assistant, you can use MQTT or the Home Assistant API. You'll need to configure your ESP32 to publish the temperature data to your Home Assistant instance using MQTT or make API requests.

6. **Update Home Assistant Configuration**: In Home Assistant, you'll need to create a sensor entity to receive the temperature data from the ESP32 and update your Home Assistant configuration accordingly.

7. **Test and Debug**: Test your setup to ensure that the temperature data is being read from the DS18B20 sensor, displayed in EMS-ESP, and sent to Home Assistant correctly. Debug any issues that may arise.

Please note that specific code and configuration details can vary depending on your setup and preferences, so you may need to consult the documentation for EMS-ESP, Home Assistant, and the libraries you're using, as well as reference examples for DS18B20 sensor integration with ESP32.

**Reply by bluhminga** [reply]
Hello,

ok, thanks, I thought there was an easier way using EMS-ESP.

**Reply by moayad-star** [reply]
If you get a sufficient answer, please mark the answer as “answered” so that it stops appearing in unanswered questions.
